regressiontests/stable/src/main/java/org/apache/isis/testdomain/conf/Configuration_headlessButSecure.java

```diff
@@ -0,0 +1,153 @@
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.apache.isis.testdomain.conf;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+import org.apache.isis.applib.annotation.OrderPrecedence;
+import org.apache.isis.applib.services.command.Command;
+import org.apache.isis.applib.services.iactn.Interaction;
+import org.apache.isis.applib.services.iactn.InteractionContext;
+import org.apache.isis.applib.services.metrics.MetricsService;
+import org.apache.isis.commons.internal.debug._Probe;
+import org.apache.isis.core.metamodel.services.command.CommandServiceInternal;
+import org.apache.isis.core.runtime.events.iactn.IsisInteractionLifecycleEvent;
+import org.apache.isis.extensions.cors.impl.IsisModuleExtCorsImpl;
+import org.apache.isis.extensions.modelannotation.metamodel.IsisModuleExtModelAnnotation;
+import org.apache.isis.security.bypass.IsisModuleSecurityBypass;
+import org.springframework.context.annotation.*;
+
+import org.apache.isis.core.runtimeservices.IsisModuleCoreRuntimeServices;
+import org.apache.isis.extensions.secman.api.SecurityModuleConfig;
+import org.apache.isis.extensions.secman.api.permission.PermissionsEvaluationService;
+import org.apache.isis.extensions.secman.api.permission.PermissionsEvaluationServiceAllowBeatsVeto;
+import org.apache.isis.testdomain.util.kv.KVStoreForTesting;
+import org.springframework.context.event.EventListener;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.TransactionDefinition;
+import org.springframework.transaction.TransactionException;
+import org.springframework.transaction.TransactionStatus;
+
+import java.util.Optional;
+
+@Configuration
+@Import({
+        IsisModuleCoreRuntimeServices.class,
+        //IsisModuleCoreSecurity.class,
+        IsisModuleSecurityBypass.class,
+        IsisModuleExtModelAnnotation.class, // @Model support
+        IsisModuleExtCorsImpl.class,
+        Configuration_headlessButSecure.HeadlessCommandSupport.class,
+        KVStoreForTesting.class, // Helper for JUnit Tests
+})
+public class Configuration_headlessButSecure {
```

Comment:
Do we need to introduce this code duplication here with respect to the existing `Configuration_headless` class? If something is preventing us from reusing `Configuration_headless`, I'd rather see `Configuration_headless` changed/fixed such that we can avoid code duplication. (This might sound picky, but its easy to add lines of code, and hard to get rid of lines of code.)